src/common/btree.c

```diff
@@ -2279,6 +2279,10 @@ btr_update(struct btr_context *tcx, d_iov_t *key, d_iov_t *val, d_iov_t *val_out
 				goto out;
 		}
 
+		/* Original record is replaced by above btr_rec_update() */
+		if (rc == 1)
```

Comment:
The `btr_node_tx_add` call may have assigned 0 to `rc`.

Reply:
oops, good catch.